optaplanner-core/src/test/java/org/optaplanner/core/impl/domain/common/accessor/MemberAccessorFactoryTest.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright 2018 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaplanner.core.impl.domain.common.accessor;
+
+import org.junit.Ignore;
+import org.junit.Test;
+import org.optaplanner.core.api.domain.solution.drools.ProblemFactProperty;
+import org.optaplanner.core.api.domain.variable.PlanningVariable;
+import org.optaplanner.core.impl.testdata.domain.TestdataValue;
+import org.optaplanner.core.impl.testdata.domain.reflect.accessmodifier.TestdataVisibilityModifierSolution;
+import org.optaplanner.core.impl.testdata.domain.reflect.field.TestdataFieldAnnotatedEntity;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertSame;
+import static org.optaplanner.core.impl.testdata.util.PlannerAssert.*;
+
+public class MemberAccessorFactoryTest {
+
+    @Test
+    public void fieldAnnotatedEntity() throws NoSuchFieldException {
+        MemberAccessor memberAccessor = MemberAccessorFactory.buildMemberAccessor(
+                TestdataFieldAnnotatedEntity.class.getDeclaredField("value"),
+                MemberAccessorFactory.MemberAccessorType.FIELD_OR_GETTER_METHOD_WITH_SETTER, PlanningVariable.class);
+        assertEquals("value", memberAccessor.getName());
+        assertEquals(TestdataValue.class, memberAccessor.getType());
```

Comment:
The test for public property includes also assertion regarding the MemberAccessor class - `assertInstanceOf(...)`. It would be worth adding it into every test method in this test case class.

And just a suggestion, I personally prefer using AssertJ for assertions instead of plain jUnit. It provides very nice messages when a test fails, has really cool capabilities when it comes to e.g. (not only) checking collections. It's already among this project's dependencies, so we can use it with every new test, if you agree.

Reply:
@yurloc and @triceo also asked to start using assertJ. There's a jira around it. I am ready to accept that, but I want to first see the junit 5 migration happen, which is 3 steps:
- Switch to the junit 5 platform runner on kie-parent (1 file change probably, very little work) and hook in junit-vintage by default.
- Migrate all optaplanner tests to junit 5. I've done the migration from 3 to 4 and it took me about a day's worth of work, by using regex tricks: https://stackoverflow.com/a/4535170/472109
- Remove junit-vintage from kie-parent, add it on all repo roots, except for optaplanner where we add junit-jupiter

Reply:
If you want to take it on:
https://issues.jboss.org/browse/PLANNER-1003

Reply:
Done assertInstanceOf() in other tests, thanks :)